api_tests/lib_sdk/actors/actor.ts

```diff
@@ -232,13 +270,66 @@ export class Actor {
             ).add(expectedBalanceChange);
             const maximumFee = wallet.MaximumFee;
 
-            const balanceInclFees = expectedBalance.sub(maximumFee);
-            const currentWalletBalance = await wallet.getBalance();
+            await expect(
+                wallet
+                    .getBalance()
+                    .then(balance =>
+                        new BigNumber(balance).gte(
+                            expectedBalance.sub(maximumFee)
+                        )
+                    )
+            ).to.eventually.be.true;
+        }
+    }
+
+    public async assertRefunded() {
+        this.logger.debug("Checking if swap @ %s was refunded", this.swap.self);
+
+        for (const [assetKind] of this.expectedBalanceChanges.entries()) {
+            const wallet = this.wallets[
+                defaultLedgerDescriptionForAsset(assetKind).name
+            ];
+            const maximumFee = wallet.MaximumFee;
 
-            expect(currentWalletBalance).to.be.gte.BN(balanceInclFees);
+            this.logger.debug(
+                "Checking that %s balance changed by max %d (MaximumFee)",
+                assetKind,
+                maximumFee
+            );
+            const expectedBalance = new BigNumber(
+                this.startingBalances.get(assetKind)
+            );
+
+            const actualBalance = new BigNumber(await wallet.getBalance());
+            await expect(actualBalance.gte(expectedBalance.sub(maximumFee))).to
+                .be.true;
         }
     }
 
+    public async assertAlphaFunded() {
+        await this.assertLedgerState("alpha_ledger", "FUNDED");
+    }
+
+    public async assertBetaFunded() {
+        await this.assertLedgerState("beta_ledger", "FUNDED");
+    }
+
+    public async assertAlphaRedeemed() {
+        await this.assertLedgerState("alpha_ledger", "REDEEMED");
+    }
+
+    public async assertBetaRedeemed() {
+        await this.assertLedgerState("beta_ledger", "REDEEMED");
+    }
+
+    public async assertAlphaRefunded() {
+        await this.assertLedgerState("alpha_ledger", "REFUNDED");
```

Comment:
```suggestion
        return this.assertLedgerState("alpha_ledger", "REFUNDED");
```

Reply:
Not sure if this makes things better.

Using `await` in a non `async` fn gives you a compile error (hence you are forced to make it an `async` fn. If you just return the promise, it is not required.

I think we get better compile-time errors if we don't return promises that make the function implicitly async.